src/articles.ts

```diff
@@ -1,13 +1,13 @@
-import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
+import { BlankNode, DatasetCore } from 'rdf-js';
 
-interface Articles extends Iterable<JsonLdObj> {
-  add(article: JsonLdObj): Promise<void>;
+interface Articles extends Iterable<[BlankNode, DatasetCore]> {
+  set(id: BlankNode, article: DatasetCore): Promise<void>;
```

Comment:
Renamed `add` to `set` as we have to separately include the ID. Makes it a bit more persistence-orientated too, and might remove the need for a unit of work when introduces a database.

Reply:
Extracted to https://github.com/libero/article-store/pull/143.